ui/panels-plugin/src/plugins/time-series-chart/time-series-chart-model.ts

```diff
@@ -37,10 +37,16 @@ export interface YAxisOptions {
   max?: number;
 }
 
+export interface PaletteOptions {
+  kind: 'Auto' | 'Categorical';
+  // colors: string []; // TODO: add colors to override ECharts theme
```

Comment:
@Nexucis so I was thinking about having a colors array (see TODO) to override the colors set in the ECharts [theme here](https://github.com/perses/perses/blob/v0.24.0/ui/components/src/utils/theme-gen.ts#L42). Do you think, we'd need a separate `kind: 'Manual'` for that? 

I was thinking, it be similar to how `abbreviate` is disabled for certain units. Basically, the ability to manually define a palette would be disabled when 'Auto' is selected, but if you pick 'Categorical', the optional color picker UI is available (but if you don't change anything it defaults to the palette set by ChartsThemeProvider).

I think the main change you are proposing is renaming 'Categorical' to 'Linear'?

Reply:
I was thinking we could have different strategy to set the color.
So here based on what you implemented I had the impression that the selection of the color is "linear" since the color number `n` is always the same.

And then maybe one day we could have a different strategy based on the label like I was explaining above.

But probably in this particular case, independently of the strategy chosen, you could define a set of colors that would be used for the `x` numbers of timeseries available (`x` is the number of colors defined).

And instead of `Auto` I would probably put `Random` if I understood correctly what it is doing.

Reply:
Okay I’m realizing from Eunice’s comment too that `getRandomColor` is a terrible function name 😂

The colors from that function aren’t actually random, they take the series name as a param, and use that to seed the color so if the same series names are used across panels, the colors should be consistent. I did fix a gap in this PR where the raw series names are used instead of the formatted series names, but otherwise I think the palette we’ve had since the beginning does meet your first comment: “same color for timeseries that are matching the same labels or the same name” 

I do think the colors that get generated aren’t great and don’t have enough contrast, but there is an adjustment we can make to improve that soon

Reply:
oh interesting ! 

What do you think about the possibility to override the first `x` timeseries color whatever the mode / strategy chosen ? 

Maybe instead of `Auto | Categorical ` we could have `Name | Linear`.
But even `Name` is not that explicite :(. It's quite hard to find a good name that summarise what it is doing 🤣. I'm so bad at naming haha